### PR TITLE
Fixes #7457: restart syslog after deleting rudder-agent.conf in report disabled mode

### PR DESCRIPTION
**Pull request by jooooooon**
https://www.rudder-project.org/redmine/issues/7457

### CODE REVIEW

File: techniques/system/common/1.0/promises.st [hunk 1]
```diff
@@ -580,14 +583,6 @@ bundle agent check_log_system
...
       "any" usebundle => rudder_common_report("Common", "result_success", "&TRACKINGKEY&", "Log system for reports", "None", "This is an android machine: Logging system configuration skipped.");
 
-  commands:
-
-    solaris.(syslog_ng_repaired|rsyslog_repaired|syslogd_repaired)::
-      "${paths.path[svcadm]} refresh svc:/system/system-log:default";
-
-    aix.syslogd_repaired::
-      "/usr/bin/refresh -s syslogd";
-
```
Comment (jooooooon) [inline techniques/system/common/1.0/promises.st]: These commands have been moved to restart_services, with the others

**Comment by jooooooon** [comment]
PR updated

**Comment by jooooooon** [comment]
PR updated

File: techniques/system/common/1.0/restart-services.st [hunk 2]
```diff
@@ -27,33 +27,43 @@
...
-    SuSE.(rsyslog_installed|rsyslog_pgsql_installed|rudder_rsyslog_conf_copied|rudder_rsyslog_pgsql|rudder_rsyslog_historical_conf_purged_repaired|syslog_ng_repaired|rsyslog_repaired|syslogd_repaired|rsyslog_limit_repaired)::
-
+    SuSE.(rsyslog_installed|rsyslog_pgsql_installed|rudder_rsyslog_conf_copied|rudder_rsyslog_pgsql|rudder_rsyslog_historical_conf_purged_repaired|syslog_ng_repaired|rsyslog_repaired|syslogd_repaired|rsyslog_limit_repaired|remove_rudder_syslog_configuration_result_repaired)::
       "restart_rsyslog" usebundle => service_restart("syslog");
-      
-    # Final report about (sys)log setting enforcement / restart
 
-    service_restart_rsyslog_repaired|service_restart_syslog_ng_repaired|service_restart_syslog_repaired::
-      "any" usebundle => rudder_common_report("Common", "log_repaired", "&TRACKINGKEY&", "Log system for reports", "None", "Logging system has been restarted");
+    (aix|solaris).(syslog_ng_repaired|rsyslog_repaired|syslogd_repaired|remove_rudder_syslog_configuration_result_repaired)::
```
Comment (peckpeck) [inline techniques/system/common/1.0/restart-services.st]: You check the change in all syslog flavours but you only restart syslod.
This seems strange.

File: techniques/system/common/1.0/restart-services.st [hunk 3]
```diff
@@ -27,33 +27,43 @@
...
     # (sys)log service
 
-    (!SuSE.!aix.!solaris).syslogd_repaired::
-
+    (!SuSE.!aix.!solaris).(syslogd_repaired|remove_rudder_syslog_configuration_syslog_repaired)::
       "restart_syslog" usebundle => service_restart("syslog");
 
-    !SuSE.syslog_ng_repaired::
-
+    !SuSE.(syslog_ng_repaired|remove_rudder_syslog_configuration_syslogng_repaired)::
```
Comment (peckpeck) [inline techniques/system/common/1.0/restart-services.st]: !SUSE includes aix and solaris, you should negate them here too

File: techniques/system/common/1.0/restart-services.st (hunk 2 above)
Comment (jooooooon) [inline techniques/system/common/1.0/restart-services.st]: Actually this just runs the unique "*syslog*" restart command, like on SuSE. On Solaris ```svcadm refresh svc:/system/system-log:default``` will restart the "currently-defined" log daemon, the same way ```service_restart("syslog")``` does on SuSE.

**Comment by jooooooon** [comment]
PR updated

**Comment by peckpeck** [comment]
Good, merging

**Pull request merged**